python/pyspark/sql/context.py

```diff
@@ -686,9 +686,10 @@ def _ssql_ctx(self):
                 self._scala_HiveContext = self._get_hive_ctx()
             return self._scala_HiveContext
         except Py4JError as e:
-            raise Exception("You must build Spark with Hive. "
-                            "Export 'SPARK_HIVE=true' and run "
-                            "build/sbt assembly", e)
+            warnings.warn("You must build Spark with Hive. "
```

Comment:
Python warnings can be disabled, so I'm worried that a lot of users might not end up seeing this message in that case. Also, this is more of an error message than a warning. Therefore, I think we should change this to a `print` statement (if we had a good logging story in PySpark, I'd say to log it as an error instead).